### PR TITLE
[Protobuf] Added static_cast

### PR DESCRIPTION
**Pull request by veblush**
Fixes https://github.com/grpc/grpc/issues/35268

### CODE REVIEW

File: include/grpcpp/support/proto_buffer_reader.h [hunk 1]
```diff
@@ -150,7 +150,7 @@ class ProtoBufferReader : public grpc::protobuf::io::ZeroCopyInputStream {
       }
       int64_t take = (std::min)(backup_count(), static_cast<int64_t>(count));
       set_backup_count(backup_count() - take);
-      count -= take;
+      count -= static_cast<int>(take);
```
Comment (yashykt) [inline include/grpcpp/support/proto_buffer_reader.h:153]: can you add a comment as to why this is safe?

File: include/grpcpp/support/proto_buffer_reader.h [hunk 2]
```diff
@@ -163,7 +163,7 @@ class ProtoBufferReader : public grpc::protobuf::io::ZeroCopyInputStream {
       set_byte_count(ByteCount() + slice_length);
       if (slice_length <= static_cast<uint64_t>(count)) {
         cord->Append(MakeCordFromSlice(grpc_slice_ref(*slice())));
-        count -= slice_length;
+        count -= static_cast<int>(slice_length);
```
Comment (yashykt) [inline include/grpcpp/support/proto_buffer_reader.h:166]: can you add a comment as to why this is safe?

**Review by yashykt — APPROVED** [review]
approved with a request for comment

File: include/grpcpp/support/proto_buffer_reader.h (hunk 2 above)
Comment (veblush) [reply include/grpcpp/support/proto_buffer_reader.h:169]: Resolved.

File: include/grpcpp/support/proto_buffer_reader.h (hunk 1 above)
Comment (veblush) [reply include/grpcpp/support/proto_buffer_reader.h:156]: resolved.

**Pull request closed without merging**